exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+}
+
+// NewOrder sends a new order to Binanceus
+func (bi *Binanceus) NewOrder(ctx context.Context, o *NewOrderRequest) (NewOrderResponse, error) {
+	var resp NewOrderResponse
+	if err := bi.newOrder(ctx, orderRequest, o, &resp); err != nil {
+		return resp, err
+	}
+	if resp.Code != 0 {
+		return resp, errors.New(resp.Msg)
+	}
+	return resp, nil
+}
+
+// NewOrderTest sends a new test order to Binanceus
+// to test new order creation and signature/recvWindow long. The endpoint creates and validates a new order but does not send it into the matching engine.
+func (bi *Binanceus) NewOrderTest(ctx context.Context, o *NewOrderRequest) (*NewOrderResponse, error) {
+	var resp NewOrderResponse
+	return &resp, bi.newOrder(ctx, testCreateNeworder, o, &resp)
+}
+
+// newOrder this endpoint is used by both new order and NewOrderTest passing their route and order information to send new order.
+func (bi *Binanceus) newOrder(ctx context.Context, api string, o *NewOrderRequest, resp *NewOrderResponse) error {
+	params := url.Values{}
+	symbol, err := bi.FormatSymbol(o.Symbol, asset.Spot)
+	if err != nil {
+		return err
+	}
+	params.Set("symbol", symbol)
+	params.Set("side", o.Side)
+	params.Set("type", string(o.TradeType))
+	if o.QuoteOrderQty > 0 {
+		params.Set("quoteOrderQty", strconv.FormatFloat(o.QuoteOrderQty, 'f', -1, 64))
+	} else {
+		params.Set("quantity", strconv.FormatFloat(o.Quantity, 'f', -1, 64))
+	}
+	if o.TradeType == BinanceRequestParamsOrderLimit {
+		params.Set("price", strconv.FormatFloat(o.Price, 'f', -1, 64))
+	}
+	if o.TimeInForce != "" {
+		params.Set("timeInForce", string(o.TimeInForce))
+	}
+	if o.NewClientOrderID != "" {
+		params.Set("newClientOrderId", o.NewClientOrderID)
+	}
+	if o.StopPrice != 0 {
+		params.Set("stopPrice", strconv.FormatFloat(o.StopPrice, 'f', -1, 64))
+	}
+	if o.IcebergQty != 0 {
+		params.Set("icebergQty", strconv.FormatFloat(o.IcebergQty, 'f', -1, 64))
+	}
+	if o.NewOrderRespType != "" {
+		params.Set("newOrderRespType", o.NewOrderRespType)
+	}
+	return bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodPost, api, params,
+		spotOrderRate, resp)
+}
+
+// GetOrder to check a trade order's status.
+func (bi *Binanceus) GetOrder(ctx context.Context, arg *OrderRequestParams) (*Order, error) {
+	var resp Order
+	params := url.Values{}
+	if arg.Symbol == "" {
+		return nil, errIncompleteArguments
+	}
+	params.Set("symbol", strings.ToUpper(arg.Symbol))
+	if arg.OrderID > 0 {
+		params.Set("orderId", strconv.Itoa(int(arg.OrderID)))
+	}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	if arg.OrigClientOrderID != "" {
+		params.Set("origClientOrderId", arg.OrigClientOrderID)
+	}
+	if arg.recvWindow > 200 && arg.recvWindow <= 6000 {
+		params.Set("recvWindow", strconv.Itoa(int(arg.recvWindow)))
+	}
+	return &resp, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodGet, orderRequest,
+		params, spotDefaultRate,
```

Comment:
The GetOrder endpoint has a ratelimit weight of 2, not 1 for _orders_. See https://docs.binance.us/#get-order